### PR TITLE
Configure ADF to checkpoint incoming bundles

### PR DESCRIPTION
**Pull request by bamader**
This PR allows the pipeline to capture pre- and post-standardization snapshots of the FHIR bundles it processes. This will allow us to do downstream analysis on the impact of our various standardization functions.

While I'm confident in the ADF JSON config file being right, I am _not_ confident in the correct way of accessing the phi storage container's URL from within the synapse notebook (I believe we need to do the snapshot modifying there as well to make before and after versions of the data LAC sends us). Dan, I pulled that (what I think is an) environment variable out of `main.tf` because I saw it as a shared variable in `outputs.tf`. That might be totally wrong though.

### CODE REVIEW

File: scripts/Synapse/convertParquetMPI.ipynb [hunk 1]
```diff
@@ -136,6 +142,17 @@
         "            'bundle': fhir_bundle,\n",
         "            'external_person_id': iris_id\n",
         "        }\n",
+        "\n",
+        "        # Store a snapshot of the FHIR bundle before it hits standardization\n",
+        "        pre_write_data = {\n",
+        "            'blob': {\"FhirResource\": fhir_bundle},\n",
+        "            'cloud_provider': 'azure',\n",
+        "            'bucket_name': 'bundle-diffs',\n",
```
Comment (DanPaseltiner) [inline scripts/Synapse/convertParquetMPI.ipynb:150]: Two thoughts here:

1. This container doesn't currently exist and this service isn't smart enough to create it for us, so well need to add terraform coverage for this. [Here](https://github.com/CDCgov/phdi-azure/blob/main/terraform/modules/shared/main.tf#L22) is an example of where we do this for the `source-data` container.
2. We aren't really storing the diffs so personally I don't think "bundle-diffs" is a good name. What about "bundle-snapshots"?

**Review by DanPaseltiner — CHANGES_REQUESTED** [review]
This is looking really great. A couple things I think we should tweak before merging.

1. As discussed lets reorder the pipeline config to make it a little easier to read
2. See my comment below on the `bundle-diffs` container

**Comment by bamader** [comment]
All comments addressed and the branch passed a successful deploy

File: terraform/modules/shared/main.tf
Comment (DanPaseltiner) [inline terraform/modules/shared/main.tf:1]: 🚀

**Review by DanPaseltiner — APPROVED** [review]
This all looks good to me. Thanks for the conversation yesterday AM and making these minor changes. Since you are out today I'm going to go ahead and merge this, as we discussed.

**Pull request merged**